staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -744,6 +745,19 @@ func (s *store) GetList(ctx context.Context, key string, opts storage.ListOption
 		})
 		metrics.RecordEtcdRequest(metricsOp, s.groupResource, err, startTime)
 		if err != nil {
+			if err == etcdrpc.ErrFutureRev {
+				requestedRV, parseErr := s.versioner.ParseResourceVersion(opts.ResourceVersion)
+				if parseErr != nil {
+					return apierrors.NewBadRequest(fmt.Sprintf("invalid resource version: %v", parseErr))
+				}
+
+				currentRV, getRVErr := s.GetCurrentResourceVersion(ctx)
+				if getRVErr != nil {
+					// If we can't get the current RV, use 0 as a fallback.
+					currentRV = 0
+				}
+				return storage.NewTooLargeResourceVersionError(requestedRV, currentRV, 1)
```

Comment:
If I remember correctly the `1` in `NewTooLargeResourceVersionError` stands for 1 second in `Retry-After`. Seems like a incorrect suggestion for clients.